kcp/setup-kcp-cps.sh

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env bash
+
+set -ex
+
+CPS_KUBECONFIG="${CPS_KUBECONFIG:-$(realpath kcp/cps-kubeconfig)}"
+WORKLOAD_KUBECONFIG="${WORKLOAD_KUBECONFIG:-$HOME/.kube/config}"
+WORKSPACE="gitops-service-$(echo $RANDOM)"
+SYNCER_IMAGE="${SYNCER_IMAGE:-ghcr.io/kcp-dev/kcp/syncer:v0.7.1}"
+SYNCER_MANIFESTS=$(mktemp -d)/cps-syncer.yaml
+
+export GITOPS_IN_KCP="true"
+
+ARGOCD_MANIFEST="$(realpath kcp/install-argocd.yaml)"
+ARGOCD_NAMESPACE="gitops-service-argocd"
+
+cleanup_workspace() {
+    pkill go
+}
+
+trap cleanup_workspace EXIT
+
+# Checks if a binary is present on the local system
+exit_if_binary_not_installed() {
+    for binary in "$@"; do
+        command -v "$binary" >/dev/null 2>&1 || {
+            echo >&2 "Script requires '$binary' command-line utility to be installed on your local machine. Aborting..."
+            exit 1
+        }
+    done
+}
+
+# Check if kubectl, kubelogin and kcp plugins are installed
+exit_if_binary_not_installed "kubectl" "kubectl-kcp" "kubectl-oidc_login"
+
+KUBECONFIG="$CPS_KUBECONFIG" kubectl config use-context kcp-stable
+
+# Opens a web browser to authenticate to your RH SSO acount
+KUBECONFIG="$CPS_KUBECONFIG" kubectl api-resources
+
+# Create a new workspace if it doesn't exist
+if KUBECONFIG="$CPS_KUBECONFIG" kubectl get workspaces "${WORKSPACE}"; then
+    echo "Workspace $WORKSPACE already exists"
+else 
+    KUBECONFIG="$CPS_KUBECONFIG" kubectl kcp ws create "$WORKSPACE"
+fi
+
+KUBECONFIG="$CPS_KUBECONFIG" kubectl kcp ws use "$WORKSPACE"
+
+KUBECONFIG="$CPS_KUBECONFIG" kubectl create ns kube-system || true
+
+# Extract the Workload Cluster name from the KUBECONFIG
+WORKLOAD_CLUSTER=$(KUBECONFIG="${WORKLOAD_KUBECONFIG}" kubectl config view --minify -o jsonpath='{.clusters[].name}' | awk -F[:] '{print $1}')
+
+WORKLOAD_CLUSTER=${WORKLOAD_CLUSTER:22}
+
+echo "Generating syncer manifests for OCP SyncTarget $WORKLOAD_CLUSTER"
+KUBECONFIG="$CPS_KUBECONFIG" kubectl kcp workload sync "$WORKLOAD_CLUSTER"  --resources "services,statefulsets.apps,deployments.apps,routes.route.openshift.io" --syncer-image "$SYNCER_IMAGE" --output-file "$SYNCER_MANIFESTS" --namespace kcp-syncer
+
+
+# Deploy the syncer to the SyncTarget
+KUBECONFIG="${WORKLOAD_KUBECONFIG}" kubectl apply -f "$SYNCER_MANIFESTS"
+
+
+echo "Waiting for the SyncTarget resource to reach ready state"
+KUBECONFIG="${CPS_KUBECONFIG}" kubectl wait --for=condition=Ready=true synctarget/"${WORKLOAD_CLUSTER}"
```

Comment:
One more thing, I had to add `wait --timeout=3m` to this line, otherwise the wait timed out. Should probably include that on this line.